libfwupdplugin/fu-usb-device.c

```diff
@@ -1017,6 +1017,9 @@ fu_usb_device_ensure_bos_descriptors(FuUsbDevice *self, GError **error)
 	if (priv->bos_descriptors_valid)
 		return TRUE;
 
+	/* clear any old descriptors */
+	g_ptr_array_set_size(priv->bos_descriptors, 0);
```

Comment:
wouldn't it make more sense to clear when setting bos_descriptors_valid to FALSE?

Reply:
Yup, we already do that in fu_usb_device_invalidate() too. I'm a bit confused why we see this failure, as if the device appears then disappears before probe completes then surely we're leaking the FuDevice as the bos_descriptors are being freed in finalize. i.e. I wonder if this is really a memory leak.

Reply:
This _feels_ like a case that `fu_device_probe_invalidate()` isn't getting called in a failure path somewhere.